### DISCUSSION TITLE
Show colored output in a browser?

### DISCUSSION THREAD

**Discussion by iliakan** [emscripten-core/emscripten#18758] [Q&A] [answered]
I'm trying to Emscripten a code that outputs colored text when running in a "regular" terminal.

How do I get colored output in browser?

Do I need to integrate using xterm_pty, xterm.js or are there other solutions?

**Comment by sbc100** [comment, marked as answer]
Emscripten doesn't provide any kind of terminal emulation.  The example page just uses a simple textarea to display output.  Anything more complex would likely involve something like xterm.js yes.  I think there are many options out there to choose from.  Another one that is very complete (IIRC) is https://hterm.org/.

**Reply by iliakan** [reply]
Nevermind, sorry, guess I wasn't clear enough, ansi-to-html can do that.

**Reply by iliakan** [reply]
Thanks for hterm reference, never heard about it before =)

**Reply by allan-bonadio** [reply]
I've got a dump function that outputs color, in JS using the usual %c directives in console.log/info/etc.  I just call the JS from C++.  It makes and throws away a Float64Array each time.
